src/script/bitcoinconsensus.cpp

```diff
@@ -75,14 +75,20 @@ static bool verify_flags(unsigned int flags)
 
 static int verify_script(const unsigned char *scriptPubKey, unsigned int scriptPubKeyLen, CAmount amount,
                                     const unsigned char *txTo        , unsigned int txToLen,
+                                    const unsigned char* spentOutputs, unsigned int spentOutputsLen,
                                     unsigned int nIn, unsigned int flags, bitcoinconsensus_error* err)
 {
     if (!verify_flags(flags)) {
         return set_error(err, bitcoinconsensus_ERR_INVALID_FLAGS);
     }
     try {
-        TxInputStream stream(PROTOCOL_VERSION, txTo, txToLen);
+        InputStream stream(PROTOCOL_VERSION, txTo, txToLen);
         CTransaction tx(deserialize, stream);
+        std::vector<CTxOut> spent_outputs;
+        if (spentOutputs != nullptr) {
+            InputStream spent_outputs_stream(PROTOCOL_VERSION, spentOutputs, spentOutputsLen);
+            spent_outputs_stream >> spent_outputs;
+        }
```

Comment:
Check the equality between inputs/outputs against `tx.vin.size()` (a `bitcoinconsensus_ERR_OUTP_INDEX` ?)